### ISSUE TITLE
Support `Route.inject()` help in koin-ktor

### ISSUE THREAD

**Issue by ValentinTrinque** [InsertKoinIO/koin#128]
*Note: I am a Kotlin beginner*

I am currently working on integrating an HTTP API using ktor, using koin-ktor. While designing my endpoints routing, I wanted to inject the dependencies at the `Route` level instead of `Application` level. Indeed, I split my `Route`s in seperate files, one for each resource :

**api/Application.kt**
```kotlin
fun Application.endpoints() {
    routing {
        v1()
    }
}

fun Routing.v1() {
    route("/api/v1") {
        users()
    }
}

```

**api/v1/Users.kt**
```kotlin
fun Route.users() {
    val usersQueryRepository: UsersQueryRepository by inject()

    get("/users") {
        val users: List<UserDto> = usersQueryRepository.list()

        call.respond(users)
    }
}
```

Your library implements `inject()` (& co.) helpers at `Application` level only. I was thinking about making this available at `Route` level.

Here is a working implementation:

```kotlin
inline fun <reified T> Route.inject(name: String = "") =
        kotlin.lazy { (StandAloneContext.koinContext as KoinContext).get<T>(name, emptyParameters()) }
```

What do you think ? I would be glad to help!

**Comment by scottsteen** [comment]
What I would rather see is the functions defined in `KoinComponent` to not be associated with an interface. This would remove the need for the `koin-ktor` project. I plan to open another issue with this request though. Kotlin actually allows you to define functions that do not belong to a class!

To get around the limitation you face is to create a Kotlin `object` that implements the `KoinComponent` interface. This way I can call `inject()` anywhere I need to.

Example:
```
object KoinProxy : KoinComponent
...
val usersQueryRepository: UsersQueryRepository by KoinProxy.inject()
...
```

Something else to consider is to call `startKoin()` in the object's `init` block. E.g.

```
object KoinProxy : KoinComponent {
    init {
        startKoin(listOf(your_context))
    }
}
```

**Comment by arnaudgiuliani** [comment]
Hello @scottsteen 

> What I would rather see is the functions defined in KoinComponent to not be associated with an interface.

This is another issue. We'll continue here about ktor Route API integration.

**Comment by r03** [comment]
For me this is not usable because the inject happens at application start and not on a http request.
I would like to create a new service at every request to that route.
For now I do the inject in every get() method instead. Is there a better way?
Maybe this is more a Ktor problem then a Koin problem.

This is the extension method I use for now:
```
inline fun <reified T> PipelineContext<Unit, ApplicationCall>.inject(name: String = "") =
        kotlin.lazy { (StandAloneContext.koinContext as KoinContext).get<T>(name, emptyParameters()) }
```

**Comment by arnaudgiuliani** [comment]
@Baresse added the extensions for `Route` & 
`Routing`. will be available in `1.0.0-beta-3`.
